pyxdsm/tests/test_xdsm.py

```diff
@@ -26,6 +25,23 @@ def tearDown(self):
         except OSError:
             pass
 
+    def test_examples(self):
+        '''
+        This test just builds the three examples, and assert that the output files exist.
+        Unlike the other tests, this one requires LaTeX to be available.
+        '''
+        os.chdir(os.path.join(os.path.dirname(os.path.abspath(__file__)), '../examples'))
+        filenames = ['kitchen_sink', 'mdf']
+        for f in filenames:
+            os.system('python {}.py'.format(f))
+            self.assertTrue(os.path.isfile(f + '.tikz'))
+            self.assertTrue(os.path.isfile(f + '.tex'))
+            self.assertTrue(os.path.isfile(f + '.pdf'))
```

Comment:
I think this should only be asserted, if pdflatex is installed, otherwise it will fail. 
Like here: https://github.com/onodip/OpenMDAO-XDSM/blob/master/omxdsm/tests/test_xdsm_viewer.py#L132

Reply:
Really good point, will borrow your approach from the XDSM plugin, thanks for pointing this out.

Reply:
Hmm should the logic be changed to:
`self.assertTrue(not pdflatex or (pdflatex and os.path.isfile(f + '.pdf')))`?
The way it's done in the referenced repo would mean that if pdflatex is installed, then it doesn't actually check `isfile` right?

Reply:
Should be equivalent.  Either `not pdflatex` is True, in this case the statement after the `or `is not even evaluated (since the PDF cannot be possibly created, it is not an error, if the file does not exist, so this is okay). If `not pdflatex` is  False - so `pdflatex `is installed, then it checks the file. In your line the or is only reached, if `pdflatex `is True, so it does not have to be checked again, that `pdflatex `is True. 

I added below a snippet, which shows, that the two approaches give the same result in all combinations. `s3 `is there to clarify the precedence.  `a=pdflatex`, `b=os.path.isfile(f + '.pdf')`



```
for a in (True, False):
    for b in (True, False):
        print(f'{a=}')
        print(f'{b=}')
        s1 = not a or b
        s2 = not a or (a and b)
        s3 = (not a) or b
        print(f'{s1=}')
        print(f'{s2=}')
        print(f'{s3=}')
        print('\n')
```

```
a=True
b=True
s1=True
s2=True
s3=True

a=True
b=False
s1=False
s2=False
s3=False

a=False
b=True
s1=True
s2=True
s3=True

a=False
b=False
s1=True
s2=True
s3=True

```

**Note:** _I found another minor logical error in the code. If pdflatex is not installed, and the file is there, the assertion is okay. This could be suspicious (maybe the file was there before the test, and therefore the test would always pass), but since a temporarily directory is used for the tests, that cannot be the case._

Reply:
Right that makes sense, thanks for that. Cheers!